src/Compilers/CSharp/Test/Semantic/Semantics/PatternMatchingTests2.cs

```diff
@@ -1093,15 +1091,15 @@ public static void M(object o)
     }
 }
 ";
-            var compilation = CreatePatternCompilation(source);
+            var compilation = CreateCompilationWithMscorlib45(source); // doesn't have ITuple
             // Two errors below instead of one due to https://github.com/dotnet/roslyn/issues/25533
             compilation.VerifyDiagnostics(
-                // (10,18): error CS1061: 'object' does not contain a definition for 'Deconstruct' and no extension method 'Deconstruct' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
+                // (8,18): error CS1061: 'object' does not contain a definition for 'Deconstruct' and no accessible extension method 'Deconstruct' accepting a first argument of type 'object' could be found (are you missing a using directive or an assembly reference?)
                 //             case (1, 2):
-                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "(1, 2)").WithArguments("object", "Deconstruct").WithLocation(10, 18),
-                // (10,18): error CS8129: No suitable Deconstruct instance or extension method was found for type 'object', with 2 out parameters and a void return type.
+                Diagnostic(ErrorCode.ERR_NoSuchMemberOrExtension, "(1, 2)").WithArguments("object", "Deconstruct").WithLocation(8, 18),
+                // (8,18): error CS8129: No suitable Deconstruct instance or extension method was found for type 'object', with 2 out parameters and a void return type.
                 //             case (1, 2):
-                Diagnostic(ErrorCode.ERR_MissingDeconstruct, "(1, 2)").WithArguments("object", "2").WithLocation(10, 18)
+                Diagnostic(ErrorCode.ERR_MissingDeconstruct, "(1, 2)").WithArguments("object", "2").WithLocation(8, 18)
                 );
```

Comment:
>); [](start = 16, length = 2)

Not related to this PR, but this reminded me: 
I'm working on an IDE PR to generate a Deconstruct method when it's missing. 
But that fixer is negatively affected by the fact that we have two diagnostics (as visible in this test). The IDE doesn't like when there are two diagnostics on the same span...
I'd like to brainstorm how to reduce this to a single diagnostic, while still providing good information to users. I'll stop by tomorrow or next week to discuss.